### PR TITLE
Testing To Do list: Testing Add and Remove Functions

### PR DESCRIPTION
**Pull request by rmiyoyo**
# Work Done
- Used jest framework for testing.
- Created a test file containing the add item and delete item functions.
- Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM.
- Created tests using the describe() method.
- Added a --coverage flag to test script in package.json. 
- Ran linter tests

### CODE REVIEW

**Comment by Developer-Kingz** [comment]
i also contributed to the project, but my name does not appear in the participant which is an issue i have been having over time. but my name can be seen in the list of commits.

**Comment by topeogunleye** [comment]
> i also contributed to the project, but my name does not appear in the participant which is an issue i have been having over time. but my name can be seen in the list of commits.

@rmiyoyo, @Developer-Kingz, please kindly check your tests for AddActivities. It's not passing:

image

File: modules/todoListActivities.test.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import AvailableActivities from './todoListActivities.js';
+
+describe('AvailableActivities', () => {
+  describe('addActivity', () => {
```
Comment (topeogunleye) [inline modules/todoListActivities.test.js:4]: - [ ] Hi, thank you for your submission. I have reviewed your code and I found some issues with your addActivity tests. They do not work as expected. Please fix these issues and resubmit your code for another review. You can read more about how to test local storage and DOM manipulation here: https://www.robinwieruch.de/local-storage-react#testing-local-storage-in-react

image

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @rmiyoyo, @Developer-Kingz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
- You have a test file ([..].test.js) for a file containing the add item and delete item functions that would be tested. This shows good organization and separation of concerns. 👏
- You group your tests using the description() method. This makes your test suite more readable and structured. 📝
- You test all functions responsible for adding and deleting list items. These are the core functionalities of your project and you cover them well with different scenarios and edge cases. 💯

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `testing-feature` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-testing-feature`. This makes it easier to track the changes and the purpose of each branch.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rmiyoyo** [comment]
# Work After First Review
image
Hi, the tests work on our side as shown in the screen. I am kindly requesting the second reviewer to let us know if there is something we are doing wrong. Thank you!

**Review by LionRouge1 — APPROVED** [review]
Hi @rmiyoyo, 

# STATUS: APPROVED! 🥇 💯 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

done

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
